test/Analysis/test-alias.mlir

```diff
@@ -29,7 +29,7 @@ tt.func @matmul_loop(%lb : index, %ub : index, %step : index, %A : !tt.ptr<f16>,
     %a = triton_gpu.convert_layout %a_ : tensor<128x32xf16, #AL> -> tensor<128x32xf16, #A_DOT>
     %b_ = tt.load %b_ptr, %b_mask, %b_other {cache = 1 : i32, evict = 1 : i32, isVolatile = false} : tensor<32x128xf16, #BL>
     %b = triton_gpu.convert_layout %b_ : tensor<32x128xf16, #BL> -> tensor<32x128xf16, #B_DOT>
-    %c = tt.dot %a, %b, %prev_c {transA = false, transB = false, allowTF32 = true, maxNumImpreciseAcc = 0 : i32} : tensor<128x32xf16, #A_DOT> * tensor<32x128xf16, #B_DOT> -> tensor<128x128xf32, #C>
+    %c = tt.dot %a, %b, %prev_c {transA = false, transB = false, f32Backend = 0 : i32, maxNumImpreciseAcc = 0 : i32} : tensor<128x32xf16, #A_DOT> * tensor<32x128xf16, #B_DOT> -> tensor<128x128xf32, #C>
```

Comment:
Is it really not possible to spell out the enumerator here, we have to use opaque numbers in the IR??

Reply:
MLIR does support printing the associated string, there is an example here:
https://github.com/llvm/llvm-project/blob/8b5b294ec2cf876bc5eb5bd5fcb56ef487e36d60/mlir/include/mlir/Dialect/GPU/IR/GPUOps.td#L40

Reply:
I tried to put the name the for quite some time before putting up the PR with little success. I think the issue is that in https://github.com/openai/triton/blob/dd8c270e5271523b9672eaa515117e6f2f5f0be6/include/triton/Dialect/Triton/IR/CMakeLists.txt#L20-L21
we are not putting these in a dialect. Then, according to [this post](https://discourse.llvm.org/t/assembly-format-of-an-array-of-enumattr/64936), we would need to refer to them as `#dialect<enum tf32>`. I tried exporting them as part of a dialect, but it broke other things.

I would like some pointers on how to do it, as I'm not fluent on MLIR.

Reply:
@ThomasRaoux do you have cycles to help out with this?  If not IMO we should land it as a string, because having magic numbers like `0` in the IR is no bueno.  (I'm willing to help, but you know a lot more about MLIR than I do...)

Reply:
Sounds like Thomas does not have time to help out with this right now.  We're OK merging as-is and figuring out how to fix this as a follow-up.